portality/models/journal.py

```diff
@@ -26,10 +26,32 @@ def find_by_issn(cls, issns, in_doaj=None, max=10):
         records = [cls(**r.get("_source")) for r in result.get("hits", {}).get("hits", [])]
         return records
 
+    @classmethod
+    def _return_one_type_issns(cls,q, type):
+        one_type = []
+        res = cls.query(q=q.query())
+        issns = [hit for hit in res.get("hits", {}).get("hits", [])[0].get("_source", {}).get("bibjson", {}).get("identifier", {})]
+        for issn in issns:
+            if issn.get("type") == type:
+                one_type.append(issn.get("id"))
+        return one_type
```

Comment:
This approach has a few problems, which might mean we re-think how we query for individual issn types:

1. It's quite computationally expensive for large result sets.  It would be better to use an aggregation, but of course we can't because of the shape of the data

2. This particular implementation only gets the first 10 results, so no matter how many journals you have, you will only see a max of 10 issns in the list.  To do it properly you'll want to page through the results, which further to (1) adds more computational expense.

I'd suggest the way to do this instead is to index e and p issns in `index.eissn` and `index.pissn` and hit the index for an aggregation on those fields, constrained by owner.  We could add those fields easily enough in `Article._generate_index()`.

If we can make it so that that all happens in one request, with both aggregations on it, we can make it much more efficient.